blaze-core/src/main/scala/org/http4s/blaze/Http1Stage.scala

```diff
@@ -234,22 +238,23 @@ object Http1Stage {
     else CachedEmptyBufferThunk
   }
 
-  /** Encodes the headers into the Writer, except the Transfer-Encoding header which may be returned
+  /** Encodes the headers into the Writer. Does not encode `Transfer-Encoding` or
+    * `Content-Length` headers, which are left for the body encoder. Adds
+    * `Date` header if one is missing and this is a server response.
+    *
     * Note: this method is very niche but useful for both server and client. */
-  def encodeHeaders(headers: Headers, rr: Writer, isServer: Boolean): Option[`Transfer-Encoding`] = {
-    var encoding: Option[`Transfer-Encoding`] = None
+  def encodeHeaders(headers: Iterable[Header], rr: Writer, isServer: Boolean) = {
```

Comment:
I don't care much about multiple traversals but I think the `filterNot` call will make a new collection which I am against. I'd also like to keep a type signature: it looks like `Unit` now but that `=` without a type before it always makes me sad. 
I don't care if this does or doesn't make it into 0.12.4 or not so if this is our favorite way to go thats fine by me.